src/codemagic/apple/app_store_connect/provisioning/beta_groups.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from codemagic.apple.app_store_connect.resource_manager import ResourceManager
+from codemagic.apple.resources import BetaGroup
+from codemagic.apple.resources import Build
+from codemagic.apple.resources import ResourceId
+from codemagic.apple.resources import ResourceType
+
+
+class BetaGroups(ResourceManager[BetaGroup]):
+    """
+    https://developer.apple.com/documentation/appstoreconnectapi/betagroup
+    """
+
+    @property
+    def resource_type(self) -> Type[BetaGroup]:
+        return BetaGroup
+
+    @dataclass
+    class Filter(ResourceManager.Filter):
+        id: Optional[Union[str, ResourceId]] = None
+        name: Optional[str] = None
+        app: Optional[str] = None
+
+    def list(self, resource_filter: Filter = Filter()):
+        """
+        https://developer.apple.com/documentation/appstoreconnectapi/list_beta_groups
+        """
+        # add a non-breakable space in the name of the automatically created internal group
+        if resource_filter.name == 'App Store Connect Users':
+            resource_filter.name = 'App\xa0Store Connect Users'
```

Comment:
I would handle it differently. In case the `\xa0` happens to be replaced by normal space in the future, then this thing blows up. I find that it is more fool-proof to list without `name` filter and then on the consumer end do predicate-based filtering over the name where both options would be allowed.